src/custom/hooks/useApproveCallback/useApproveCallbackMod.ts

```diff
@@ -43,14 +52,19 @@ export function useApproveCallback(
     // we might not have enough data to know whether or not we need to approve
     if (!currentAllowance) return ApprovalState.UNKNOWN
 
+    // is the optionally set check amount less than current set allowance?
+    const isOptionalAmountLessThanAllowance = !!amountToCheckAgainstAllowance?.lessThan(currentAllowance)
```

Comment:
Why the `!!`? Isn't it a boolean in all cases?

Reply:
i think is because of that `?`, cause can be undefined